### DISCUSSION TITLE
Show Loading While Waiting For Diagram Display

### DISCUSSION THREAD

**Discussion by QuangLanh** [clientIO/joint#2582] [Q&A]
### Introduction

I want when the user is waiting for the diagram to display, display loading on the interface so the user knows the system is loading

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by QuangLanh** [comment]
please reply me early @@

**Comment by kumilingus** [comment]
Why does the user need to wait? Are you fetching the diagram from the database?

Or does it take a long time to render the diagram?

If you want to show a loader with progress during the rendering you can use `progress` callback of paper's `unfreeze()` method (more info in [docs](https://resources.jointjs.com/docs/jointjs/v4.0/joint.html#dia.Paper.prototype.unfreeze)).

See this [example](https://changelog.jointjs.com/rappid/examples/perf/). Set `batchSize` to e.g. `1` and hit `restart`.
The source code for the example is [here](https://github.com/clientIO/joint/blob/master/packages/joint-core/demo/performance/async.js#L225-L240).

**Reply by QuangLanh** [reply]
Because I saved a big diagram in the database and it takes a lot of time to render that diagram. So, while users are waiting for the diagram to appear, we need to show a loading message to let them know.

**Reply by kumilingus** [reply]
Please define a big diagram (how many elements and cells).

If you are sure that the rendering of the objects causing the delay and not the network request, you can use the technique from the example above.

**Reply by QuangLanh** [reply]
I will try it
Thank you so much.
